SourceCode/raytracer/CRTTypes.h

```diff
@@ -245,5 +245,75 @@ class Triangle {
         return "Triangle: {" + v0.toString() + ", " + v1.toString() + ", " + v2.toString() + "}";
     }
 
+    /*
+    * Input: ray, rayProj
+    * Input: rayProj: projection of ray.direction onto camera direction
+    * Output: t: distance from ray.origin to the intersection point
+    * Output: p: intersection point
+    * return: true if the ray intersects the triangle
+    */
+    bool intersect(const Ray& ray, float& t, Vec3& p) const {
```

Comment:
my pdf notes refer to intersect_plane and intersect_plane2, which are a simplified version of intersect